test/integration/test_dataset.py

```diff
@@ -34,13 +34,32 @@ def test_populate(conservator):
     assert dataset_from_id.name == dataset.name
 
 
-@pytest.mark.skip()
 def test_delete(conservator):
     dataset = conservator.datasets.create("My dataset")
-    assert conservator.datasets.count() == 1
+    # Give the server time to create and commit the dataset.
```

Comment:
I don't think that this section from 39-49 actually does anything useful; once the `conservator.datasets.create()` returns, it returns the dataset with an ID, which means that it is in conservator. 

The check that really needs to happen is to poll the dataset by ID and check it's commit status field

Reply:
Added a new commit that waits on the git_commit_state field to change to "completed" before checking out or deleting the dataset.